pylabrobot/plate_reading/biotek_backend.py

```diff
@@ -0,0 +1,250 @@
+    assert resp2 == b"0000\x03"
+
+    # read data
+    body = await self._read_until(b"\x03")
+    assert resp is not None
+    return self._parse_body(body)
+
+  async def read_luminescence(self, focal_height: float) -> List[List[float]]:
+    raise NotImplementedError("Not implemented yet")
+
+  async def read_fluorescence(
+    self,
+    excitation_wavelength: int,
+    emission_wavelength: int,
+    focal_height: float,
+  ) -> List[List[float]]:
+    if not 4.5 <= focal_height <= 13.88:
+      raise ValueError("Focal height must be between 4.5 and 13.88")
+    if not 250 <= excitation_wavelength <= 700:
+      raise ValueError("Excitation wavelength must be between 250 and 700")
+    if not 250 <= emission_wavelength <= 700:
+      raise ValueError("Emission wavelength must be between 250 and 700")
+
+    resp = await self.send_command("t", wait_for_char=b"\x06")
+    assert resp == b"\x06"
+
+    cmd = f"{614220 + int(1000*focal_height)}\x03".encode()
+    await self.send_command(cmd, purge=False)
+
+    resp = await self.send_command("y", wait_for_char=b"\x06")
+    assert resp == b"\x06"
+    await self.send_command(b"08120112207434014351135308559127881772\x03", purge=False)
+
+    resp = await self.send_command("D", wait_for_char=b"\x06")
+    assert resp == b"\x06"
+    excitation_wavelength_str = str(excitation_wavelength).zfill(4)
+    emission_wavelength_str = str(emission_wavelength).zfill(4)
+    cmd = (f"008401010108120001200100001100100000135000100200200{excitation_wavelength_str}000"
+      f"{emission_wavelength_str}000000000000000000210011").encode()
+    checksum = str((sum(cmd)+7) % 100).encode() # don't know why +7
+    cmd = cmd + checksum + b"\x03"
+    await self.send_command(cmd, purge=False)
+
+    resp1 = await self.send_command("O", wait_for_char=b"\x06")
+    assert resp1 == b"\x06"
+    resp2 = await self._read_until(b"\x03")
+    assert resp2 == b"0000\x03"
+
+    # read data
+    body = await self._read_until(b"\x03", timeout=60*2)
+    assert body is not None
+    return self._parse_body(body)
+
+  async def _abort(self) -> None:
+    await self.send_command("x", wait_for_char=None)
+
+  class ShakeType(enum.Enum):
+    LINEAR = "linear"
+    ORBITAL = "orbital"
+
+  async def shake(self, shake_type: ShakeType) -> None:
+    """ Warning: the duration for shaknig has to be specified on the machine, and the maximum is
+    16 minutes and 39 seconds. As a hack, we start shaking for the maximum duration every time
+    as long as stop is not called. """
+    max_duration = 16*60 + 39 # 16m39s
+
+    async def shake_maximal_duration():
+      resp = await self.send_command("y", wait_for_char=b"\x06")
+      assert resp == b"\x06"
+      await self.send_command(b"08120112207434014351135308559127881422\x03", purge=False)
+
+      resp = await self.send_command("D", wait_for_char=b"\x06")
+      assert resp == b"\x06"
+      shake_type_bit = "0" if shake_type == self.ShakeType.LINEAR else "1"
```

Comment:
```suggestion
      shake_type_bit = str(self.ShakeType.LINEAR.value)
```
and make ShakeType an IntEnum?